src/components/AutoComplete/useGetItems.tsx

```diff
@@ -35,7 +37,8 @@ function useGetItems<Item>(
           inputValue: changed?.inputValue,
         });
 
-        const items = (data || []).filter(Boolean) as Item[]; // 🚨 Make sure we don't have array of null (ie. null[])
+        const apiItems = hasItemsFooter ? [...data, { id: footerId }] : data;
+        const items = (apiItems || []).filter(Boolean) as Item[]; // 🚨 Make sure we don't have array of null (ie. null[])
```

Comment:
Given we are now using `emptyItem` is this check still required?

Reply:
apiItems can be undefined and we don't have control over it. So I'd have the safeguard here to make sure the code doesn't break